core/src/main/java/tc/oc/pgm/variables/types/WorldTimeVariable.java

```diff
@@ -0,0 +1,24 @@
+package tc.oc.pgm.variables.types;
+
+import tc.oc.pgm.api.match.Match;
+
+public class WorldTimeVariable extends AbstractVariable<Match> {
+
+  public static final WorldTimeVariable INSTANCE = new WorldTimeVariable();
+
+  public WorldTimeVariable() {
+    super(Match.class);
+  }
+
+  @Override
+  protected double getValueImpl(Match match) {
+    long val = match.getWorld().getFullTime();
+    return (double) val;
+  }
+
+  @Override
+  protected void setValueImpl(Match match, double value) {
+    long val = (long) value;
+    match.getWorld().setFullTime(val);
```

Comment:
From: https://jd.papermc.io/paper/1.21.5/org/bukkit/World.html#setFullTime(long)

> void setFullTime(long time)
Sets the in-game time on the server
Note that this sets the full time of the world, which may cause adverse effects such as breaking redstone clocks and any scheduled events

Using `setTime` and correspondingly `getTime` seems like it would be safer